File: 2012 Book Source Code/Chapter_7_Source_Code/Chapter 7/Employee Web Service VB/Employee.asmx.vb
Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports System.ComponentModel
Imports System.Data
Imports System.Web.Configuration
Imports System.Data.SqlClient

<System.Web.Services.WebService(Namespace:="http://tempuri.org/")> _
<System.Web.Services.WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<ToolboxItem(False)> _
Public Class Employee
    Inherits System.Web.Services.WebService

    <WebMethod()> _
    Public Function GetCost(ByVal employeeID As String, ByVal visitDate As DateTime) As Decimal

        Dim myConnection As New SqlConnection()
        myConnection.ConnectionString = WebConfigurationManager.ConnectionStrings("Pro_SSRS").ConnectionString

        Dim queryString As String = "EmpVisitCost"

        Dim command As New SqlCommand(queryString, myConnection)
        command.CommandType = CommandType.StoredProcedure
        command.Parameters.Add("@EmployeeID", SqlDbType.Char, 20).Value = employeeID
        command.Parameters.Add("@VisitDate", SqlDbType.DateTime).Value = visitDate

        Dim parameter As SqlParameter = New SqlParameter("@VisitCost", SqlDbType.Decimal)
        parameter.Direction = ParameterDirection.Output
        command.Parameters.Add(parameter)

        myConnection.Open()
        command.ExecuteNonQuery()

        Return Decimal.Parse(parameter.Value)

    End Function

End Class

File: 2012 Book Source Code/Chapter_7_Source_Code/Chapter 7/Employee Web Service/Employee.asmx.cs
using System;
using System.Web;
using System.Web.Services;
using System.Web.Services.Protocols;
using System.Data;
using System.Web.Configuration;
using System.Data.SqlClient;

namespace Employee_Web_Service
{
    /// <summary>
    /// Summary description for Service1
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    public class Employee : System.Web.Services.WebService
    {

        [WebMethod]
        public decimal GetCost(string employeeID, DateTime visitDate)
        {

            SqlConnection myConnection = new SqlConnection();
            myConnection.ConnectionString = WebConfigurationManager.ConnectionStrings["Pro_SSRS"].ConnectionString;
            
            string queryString = "EmpVisitCost";
            SqlCommand command = new SqlCommand(queryString, myConnection);
            
            command.CommandType = CommandType.StoredProcedure;
            command.Parameters.Add("@EmployeeID", SqlDbType.Char, 20).Value = employeeID;
            command.Parameters.Add("@VisitDate", SqlDbType.DateTime).Value = visitDate;
            
            SqlParameter parameter = new SqlParameter("@VisitCost", SqlDbType.Decimal);
            parameter.Direction = ParameterDirection.Output;
            command.Parameters.Add(parameter);
            
            myConnection.Open();
            command.ExecuteNonQuery();
            
            decimal cost = (decimal)parameter.Value;
            return cost;

        }
    }
}